src/core.ts

```diff
@@ -70,15 +74,32 @@ export type Observable<T> = {
 export type FromInput<T> = Promise<T> | Stream<T> | Array<T> | Observable<T>;
 
 // mutates the input
-function internalizeProducer<T>(producer: Producer<T>) {
+function internalizeProducer<T>(producer: Producer<T> | ProducerFn<T>) {
```

Comment:
My personal feeling on this is that it should be a decision on either an object `{ start, stop }` *or* `(listener) => disposeFn`. Having both is a complication to the overall design IMO.